src/imagery/i.eodag/i.eodag.py

```diff
@@ -468,8 +477,14 @@ def no_fallback_search(search_parameters, provider):
     :rtype: class:'eodag.api.search_result.SearchResult'
     """
     try:
-        server_poke = dag.search(**search_parameters, provider=provider)
-        if server_poke[1] == 0:
+        server_poke = dag.search(**search_parameters, provider=provider, count=True)
+        if int(eodag.__version__.split(".")[0]) < 3 and server_poke[1] == 0:
+            gs.verbose(_("No products found"))
+            return SearchResult([])
+        if (
+            int(eodag.__version__.split(".")[0]) >= 3
+            and server_poke.number_matched == 0
+        ):
```

Comment:
This can be combined and shortened in one if condition:
```suggestion
        if (int(eodag.__version__.split(".")[0]) < 3 and server_poke[1] == 0) or
        (
            int(eodag.__version__.split(".")[0]) >= 3
            and server_poke.number_matched == 0
        ):
```